testsuite/manualmode/src/test/java/org/wildfly/core/test/standalone/mgmt/PreparedResponseTestCase.java

```diff
@@ -0,0 +1,152 @@
+        helper.deploy(SLOW_STOP_JAR, war.as(ZipExporter.class).exportAsInputStream());
+    }
+
+    public static void undeploy(ManagementClient managementClient) throws ServerDeploymentHelper.ServerDeploymentException {
+        ServerDeploymentHelper helper = new ServerDeploymentHelper(managementClient.getControllerClient());
+        helper.undeploy(SLOW_STOP_JAR);
+    }
+
+    @Test
+    public void reloadServer() throws Exception {
+        ManagementClient mgmtClient = getManagementClient();
+        long timeout = SHUTDOWN_WAITING_TIME + System.currentTimeMillis();
+        mgmtClient.executeForResult(Operations.createOperation(RELOAD, new ModelNode().setEmptyList()));
+        while (System.currentTimeMillis() < timeout) {
+            Thread.sleep(FREQUENCY);
+            try {
+                controller.getClient().isServerInRunningState();
+            } catch (RuntimeException ex) {
+                ex.printStackTrace();
+                break;
+            }
+        }
+        Assert.assertFalse(System.currentTimeMillis() < timeout);
+        Thread.sleep(2 * FREQUENCY);
+        Assert.assertTrue(mgmtClient.isServerInRunningState());
+    }
+
+    @Test
+    public void shutdownServer() throws Exception {
+        try (ManagementClient managementClient = getManagementClient()) {
+            long timeout = SHUTDOWN_WAITING_TIME + System.currentTimeMillis();
+            managementClient.executeForResult(Operations.createOperation(SHUTDOWN, new ModelNode().setEmptyList()));
+            while (System.currentTimeMillis() < timeout) {
+                Thread.sleep(FREQUENCY);
+                try {
+                    controller.getClient().isServerInRunningState();
+                } catch (RuntimeException ex) {
+                    ex.printStackTrace();
+                    break;
+                }
+            }
+            Assert.assertFalse(System.currentTimeMillis() < timeout);
+            Assert.assertFalse(controller.getClient().isServerInRunningState());
+        }
+        Assert.assertFalse(controller.getClient().isServerInRunningState());
+        try {
+            controller.stop(); //Server is already stopped but the ServerController doesn't know it
+        } catch (RuntimeException ex) {
+            //ignore the exception as it is expected
+        }
+        controller.start();
+    }
+
+    static ManagementClient getManagementClient() {
+        ModelControllerClient client = null;
+        try {
+            client = ModelControllerClient.Factory.create("http-remoting", InetAddress.getByName(TestSuiteEnvironment.getServerAddress()),
+                    MANAGEMENT_HTTP_PORT, new org.wildfly.core.testrunner.Authentication.CallbackHandler());
+        } catch (UnknownHostException e) {
+            throw new RuntimeException(e);
+        }
+        return new ManagementClient(client, TestSuiteEnvironment.getServerAddress(), MANAGEMENT_HTTP_PORT, "http-remoting");
```

Comment:
This should just be
```java
return new ManagementClient(TestSuiteEnvironment.getModelControllerClient(), TestSuiteEnvironment.getServerAddress(), TestSuiteEnvironment.getServerPort(), "http-remoting");
```
It probably shouldn't be using `org.jboss.as.test.integration.management.util.CustomCLIExecutor.MANAGEMENT_HTTP_PORT` at all since this isn't really a CLI test.